src/widgets/PopoverDropDown.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable react/forbid-prop-types */
+
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withNavigation } from 'react-navigation';
+import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
+import Popover from 'react-native-popover-view';
+
+import { usePopover } from '../hooks';
+
+import {
+  SUSSOL_ORANGE,
+  BLUE_WHITE,
+  SHADOW_BORDER,
+  APP_FONT_FAMILY,
+  textStyles,
+  APP_GENERAL_FONT_SIZE,
+} from '../globalStyles';
+
+const DropDownOption = ({ text, onPress, isLast }) => {
+  const {
+    dropDownOptionText,
+    dropDownOptionContainer,
+    dropDownOptionInnerContainer,
+    dropDownOptionInnerBorder,
+  } = localStyles;
+
+  return (
+    <View style={dropDownOptionContainer}>
+      <View
+        style={{ ...dropDownOptionInnerContainer, ...(isLast ? {} : dropDownOptionInnerBorder) }}
+      >
+        <TouchableOpacity onPress={onPress}>
+          <Text style={dropDownOptionText}>{text}</Text>
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+};
+
+DropDownOption.propTypes = {
+  text: PropTypes.string.isRequired,
+  onPress: PropTypes.func.isRequired,
+  isLast: PropTypes.bool.isRequired,
+};
+
+const DropDown = ({ anchorRef, isVisible, options, onSelection, onClose, title }) => {
+  const {
+    dropDownContainer,
+    dropDownTitleContainer,
+    dropDownTitleInnerContainer,
+    popoverBackgroundStyle,
+    arrowStyle,
+  } = localStyles;
+
+  return (
+    <Popover
+      isVisible={isVisible}
+      fromView={anchorRef.current}
+      onRequestClose={onClose}
+      arrowStyle={arrowStyle}
+      backgroundStyle={popoverBackgroundStyle}
+      placement="bottom"
+      animationConfig={{ duration: 50 }}
+    >
+      <View style={dropDownContainer}>
+        <View style={dropDownTitleContainer}>
+          <View style={dropDownTitleInnerContainer}>
+            <Text style={textStyles}>{title}</Text>
+          </View>
+        </View>
+        {options.map((option, index) => (
+          <DropDownOption
+            text={option}
+            onPress={onSelection}
+            isLast={options.length - 1 === index}
+          />
+        ))}
+      </View>
+    </Popover>
+  );
+};
+
+export const PopoverDropDown = withNavigation(
+  ({ navigation, BaseComponent, options, onSelection }) => {
+    const [ref, visible, show, close] = usePopover(navigation);
+    return (
+      <>
+        <TouchableOpacity onPress={show} ref={ref}>
+          <BaseComponent />
+        </TouchableOpacity>
+        <DropDown
+          anchorRef={ref}
+          options={options}
+          isVisible={visible}
+          onSelection={onSelection}
+          onClose={close}
+          title="Payment Types"
```

Comment:
would the props for PopoverDropDown be coming later? The title, options and onSelection are set by the caller aren't they?

Reply:
oops - have added the title prop 1409643  - yeah the props you've mentioned would be supplied, as well as a `BaseComponent` to act as an anchor.

Have added a docstring.,. hoping that it's enough even though I think I know it's not